.buildkite/pipeline.yml

```diff
@@ -1,3 +1,66 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+
+env:
+  SETUP_GVM_VERSION: 'v0.5.0'
+  DOCKER_COMPOSE_VERSION: '1.25.5'
+  ELASTIC_PACKAGE_COMPOSE_DISABLE_ANSI: "true"
+  KIND_VERSION: 'v0.17.0'
+  K8S_VERSION: 'v1.26.0'
+
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - group: "Check and build"
+    key: "tests"
+    steps:
+      - label: ":go: Run check-ci"
+        key: check-ci
+        command: "make check-ci"
+        agents:
+          image: "golang:1.20.3"
+          cpu: "8"
+          memory: "4G"
+
+      - label: ":go: Run local build"
+        key: make-local
+        command: "make local"
+        agents:
+          image: "golang:1.20.3"
+          cpu: "8"
+          memory: "4G"
+
+  - group: "Run tests"
+    steps:
+      - label: "Run unit test"
+        key: unit-test
+        commands:
+          - "chmod +x .buildkite/scripts/setenv.sh"
+          - ".buildkite/scripts/setenv.sh with-go"
+          - "make test-unit"
+          - "make junit-report"
+          - "buildkite-agent artifact upload build/*.xml"
+        agents:
+          provider: "gcp"
+
+
+      - label: "Run integration tests"
+        key: int-test
+        commands:
+          - "chmod +x .buildkite/scripts/setenv.sh"
+          - ".buildkite/scripts/setenv.sh with-go-docker-compose"
+          - "make test-int"
+          - "make junit-report"
+          - "buildkite-agent artifact upload build/*.xml"
+        agents:
+          provider: "gcp"
+
+      - label: ":junit: Junit annotate"
```

Comment:
Is this step executed when unit test or integration test fail ? 

Not sure if it is needed to change the `depends_on` relationship, to a specific step to wait and continue with failure.

For instance, add this step between the unit/integration test and the junit report

```yaml
      - wait: ~
        continue_on_failure: true
```

Reply:
What it is interesting is to run the junit step even if any of the unit or integration test steps fail to be able to show the failed tests. Allowing that, junit step could add the annotation for the tests failed in the UI.

Would that relationship allow that ?

Reply:
Ok, I see that you have added `allow_failure`  :+1: 
```
        depends_on:
          - step: "unit-test"
            allow_failure: true
          - step: "int-test"
            allow_failure: true
```